src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java

```diff
@@ -945,6 +945,72 @@ public void testBulkInsertWithAllTemporalTypesAndMoneyAsVarchar() throws Excepti
         }
     }
 
+    /**
+     * Test bulk insert with null data for all temporal types and money as varchar when useBulkCopyForBatchInsert is true.
+     * sendTemporalDataTypesAsStringForBulkCopy is set to true by default.
+     * Temporal types are sent as varchar, and money/smallMoney are sent as their respective types.
+     * 
+     * @throws Exception
+     */
+    @Test
+    @Tag(Constants.xAzureSQLDW)
+    public void testBulkInsertWithNullDataForAllTemporalTypesAndMoneyAsVarchar() throws Exception {
+        String tableName = RandomUtil.getIdentifier("BulkTable");
+        String createTableSQL = "CREATE TABLE " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (" +
+                "dateTimeColumn DATETIME, " +
+                "smallDateTimeColumn SMALLDATETIME, " +
+                "dateTime2Column DATETIME2, " +
+                "dateColumn DATE, " +
+                "timeColumn TIME, " +
+                "dateTimeOffsetColumn DATETIMEOFFSET, " +
+                "moneyColumn MONEY, " +
+                "smallMoneyColumn SMALLMONEY" + ")";
```

Comment:
Common code can be moved to some method.  like createTableSQL is same in both test cases.